Skip a warning using EC_GFp_nist_method() which was removed in LibreSSL 3.8

Index: src/lib/tls/tortls_openssl.c
--- src/lib/tls/tortls_openssl.c.orig
+++ src/lib/tls/tortls_openssl.c
@@ -341,7 +341,8 @@ tor_tls_init(void)
 #endif /* defined(OPENSSL_1_1_API) */
 
 #if (SIZEOF_VOID_P >= 8 &&                              \
-     OPENSSL_VERSION_NUMBER >= OPENSSL_V_SERIES(1,0,1))
+     OPENSSL_VERSION_NUMBER >= OPENSSL_V_SERIES(1,0,1) && \
+     !defined(LIBRESSL_VERSION_NUMBER))
     long version = tor_OpenSSL_version_num();
 
     /* LCOV_EXCL_START : we can't test these lines on the same machine */
